scripts/hid_configurator/gui/gui.py

```diff
@@ -0,0 +1,159 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+
+import threading
+from kivy.app import App
+from kivy.clock import mainthread
+from kivy.config import Config
+from kivy.properties import ObjectProperty
+from kivy.uix.button import Button
+from kivy.uix.floatlayout import FloatLayout
+from kivy.uix.gridlayout import GridLayout
+from kivy.uix.popup import Popup
+
+from gui_backend import DEVICE_PID, NORDIC_VID, DEVICE_TYPE, Device
+
+GENERIC_DESKTOP_PAGE = 1
+
+
+class MouseOptions(GridLayout):
+    pass
+
+
+class DfuButtons(GridLayout):
+    def start_dfu_thread(self):
+        app_ref = App.get_running_app()
+        threading.Thread(target=app_ref.dfu, args=(self.update_progressbar,)).start()
+
+    @mainthread
+    def update_progressbar(self, permil):
+        self.ids.pb.value = permil / 10 + 0.1
```

Comment:
maybe we could sleep here (update only periodically) ?

Reply:
Discussed offline